src/ota_stm32h7.c

```diff
@@ -121,6 +124,12 @@ MG_IRAM static bool mg_stm32h7_swap(void) {
   flash_clear_err(bank);
   // printf("OPTSR_PRG 1 %#lx\n", FLASH->OPTSR_PRG);
   MG_SET_BITS(MG_REG(bank + FLASH_OPTSR_PRG), MG_BIT(31), desired);
+  if (STM_DEV_ID == 0x450) {
+    // H745/H755 and H747/H757 are running on dual core
+    // Disable CM4 to prevent the old and the new firmwares from booting
+    // simultaneously on each core.
+    MG_SET_BITS(MG_REG(bank + FLASH_OPTSR_PRG), MG_BIT(22), 0);
+  }
```

Comment:
This is quick workaround, not a fix...
The user bought a dual-core chip for a reason, and we are rendering the second core unusable after first upgrade.
The problem with dual-core OTA is that any core firmware might want to be updated.
IMO, at a minimum, we should be able to update the core we are running in without disabling the other core.
Ideally, we should be able to update both cores, but I guess that would require a different schema, as each core has its own image.

Reply:
To answer your questions,

Yes, we can change the base addresses for each core. There are some option bytes which control this and modifying the values seems to be persistent across reboots. I tried to figure out how to do it in Cube, but I can't find such an option. Cube seems to generate code linked at the default addresses for each core. (CM4 code is linked at 0x81.... and CM7 at 0x80...). You can change these option bytes though in STM32CubeProgrammer, but I think that is irrelevant for now.

For now, I don't think we should deal with M4 memory space. As long as we use the first bank only for OTA, everything should be fine, since we are updating only M7.

Reply:
Exactly! We'll treat it as a single bank scenario by assuming the 2nd bank does not exist.